code/game/objects/structures/door_assembly.dm

```diff
@@ -703,38 +493,66 @@
 					door.req_access = electronics.accesses
 				if(created_name)
 					door.name = created_name
+				else
+					door.name = base_name
+				door.previous_airlock = previous_assembly
 				electronics.forceMove(door)
 				qdel(src)
 	else
 		return ..()
+	update_name()
 	update_icon()
 
 /obj/structure/door_assembly/update_icon()
 	cut_overlays()
-	if(!material)
+	if(!glass)
 		add_overlay(get_airlock_overlay("fill_construction", icon))
-	else
-		add_overlay(get_airlock_overlay("[material]_construction", overlays_file))
+	else if(glass)
+		add_overlay(get_airlock_overlay("glass_construction", overlays_file))
 	add_overlay(get_airlock_overlay("panel_c[state+1]", overlays_file))
 
+/obj/structure/door_assembly/proc/update_name()
+	name = ""
+	switch(state)
+		if(AIRLOCK_ASSEMBLY_NEEDS_WIRES)
+			if(anchored)
+				name = "secured "
+		if(AIRLOCK_ASSEMBLY_NEEDS_ELECTRONICS)
+			name = "wired "
+		if(AIRLOCK_ASSEMBLY_NEEDS_SCREWDRIVER)
+			name = "near finished "
+	name += "[heat_proof_finished ? "heat-proofed " : ""][glass ? "window " : ""][base_name] assembly"
+
+/obj/structure/door_assembly/proc/transfer_assembly_vars(obj/structure/door_assembly/source, obj/structure/door_assembly/target, previous = FALSE)
+	target.glass = source.glass
+	target.heat_proof_finished = source.heat_proof_finished
+	target.created_name = source.created_name
+	target.state = source.state
+	target.anchored = source.anchored
+	if(previous)
+		target.previous_assembly = source.type
+	if(electronics)
+		target.electronics = source.electronics
+		source.electronics.forceMove(target)
+	target.update_icon()
+	target.update_name()
+	qdel(source)
 
 /obj/structure/door_assembly/deconstruct(disassembled = TRUE)
 	if(!(flags_1 & NODECONSTRUCT_1))
 		var/turf/T = get_turf(src)
-		var/metal_amt = 4
 		if(!disassembled)
-			metal_amt = rand(2,4)
-		new /obj/item/stack/sheet/metal(T, metal_amt)
-		if(mineral)
-			if (mineral == "glass")
-				if(disassembled)
-					if (heat_proof_finished)
-						new /obj/item/stack/sheet/rglass(T)
-					else
-						new /obj/item/stack/sheet/glass(T)
+			material_amt = rand(2,4)
+		new material_type(T, material_amt)
+		if(glass)
+			if(disassembled)
+				if(heat_proof_finished)
+					new /obj/item/stack/sheet/rglass(T)
 				else
-					new /obj/item/shard(T)
+					new /obj/item/stack/sheet/glass(T)
 			else
-				var/obj/item/stack/sheet/mineral/mineral_path = text2path("/obj/item/stack/sheet/mineral/[mineral]")
-				new mineral_path(T, 2)
+				new /obj/item/shard(T)
+		if(mineral)
+			var/obj/item/stack/sheet/mineral/mineral_path = text2path("/obj/item/stack/sheet/mineral/[mineral]")
```

Comment:
Any way to get rid of these while you're refactoring the whole thing ?

Reply:
I will take a look at it and ask on coderbus over the next few days, no guarantees though as I'm kind of a noob coder

Reply:
Alright, can wait for another PR though.